serde-generate/src/generate.rs

```diff
@@ -20,29 +20,88 @@ enum Language {
 }
 }
 
+arg_enum! {
+#[derive(Debug, StructOpt, PartialEq, Eq, PartialOrd, Ord)]
+enum Runtime {
+    Serde,
+    Bincode,
+}
+}
+
 #[derive(Debug, StructOpt)]
 #[structopt(
     name = "Serde code generator",
     about = "Generate code for Serde containers"
 )]
 struct Options {
     #[structopt(parse(from_os_str))]
-    input: PathBuf,
+    input: Option<PathBuf>,
 
     #[structopt(long, possible_values = &Language::variants(), case_insensitive = true, default_value = "Python3")]
     language: Language,
+
+    #[structopt(long)]
+    with_runtimes: Vec<Runtime>,
+
+    #[structopt(long)]
+    name: Option<String>,
+
+    #[structopt(long)]
+    target_source_dir: Option<PathBuf>,
 }
 
 fn main() {
     let options = Options::from_args();
-    let content =
-        std::fs::read_to_string(options.input.as_os_str()).expect("input file must be readable");
-    let registry = serde_yaml::from_str::<Registry>(content.as_str()).unwrap();
+    let named_registry_opt = match &options.input {
+        None => None,
+        Some(input) => {
+            let name = options.name.clone().unwrap_or_else(|| {
+                input
+                    .file_stem()
+                    .expect("failed to deduce module name from input path")
+                    .to_string_lossy()
+                    .to_string()
+            });
+            let source_path = input.as_os_str();
+            let content =
+                std::fs::read_to_string(source_path).expect("input file must be readable");
+            let registry = serde_yaml::from_str::<Registry>(content.as_str()).unwrap();
+            Some((registry, name))
+        }
+    };
+
+    match options.target_source_dir {
+        None => {
+            if let Some((registry, _)) = named_registry_opt {
+                let mut out = std::io::stdout();
```

Comment:
For performance you may want to lock stdout (assuming you don't need to write out to stdout anywhere else)